### PR TITLE
More minor fixes

### PR DESCRIPTION
**Pull request by unclecheese**
* Fix `Folder::find_or_make` bug where root `assets/` folder was not being removed.
* Tidy up JS functions

### CODE REVIEW

File: code/AssetGalleryField.php [hunk 1]
```diff
@@ -216,7 +216,7 @@ protected function getData($filters = array()) {
 	 */
 	protected function getFolder($folder = null) {
 		if ($folder) {
-			return Folder::find_or_make($folder);
+			return Folder::find_or_make(preg_replace('/^'.ASSETS_DIR.'\//', '', $folder));
```
Comment (assertchris) [inline code/AssetGalleryField.php]: Would something like `str_ireplace` be simpler here?

File: public/src/store/editorStore.js [hunk 2]
```diff
@@ -6,9 +6,7 @@ import CONSTANTS from '../constants';
 var _fields = [];
 
 function create(data) {
-	var fieldExists = _fields.filter((field) => { return field.name === data.name; }).length > 0;
-
-	if (fieldExists) {
+	if(_fields.some(field => field.name === data.name)) {
```
Comment (assertchris) [inline public/src/store/editorStore.js]: <3

**Comment by assertchris** [comment]
Oh! I just read the description of this. I'm not sure we want the assets path removed - that makes it impossible to browse from that folder, which means all uploads need to be in a subfolder. @tractorcow opinions?

**Comment by tractorcow** [comment]
Yep, we don't want any references to assets at all, really.

`Folder::find_or_make` will trust any data it's given. Case in point, it should be perfectly valid to create an `assets/assets` folder.

**Comment by tractorcow** [comment]
Also, the "Filename" property of File DataObject will no longer contain the assets folder anyway, so even fewer reasons to care about it. ;D

**Pull request closed without merging**